src/components/ColumnAdjuster/ColumnAdjuster.tsx

```diff
@@ -0,0 +1,135 @@
+ * When you start dragging, mouse move and mouse up listeners are added (as well as corresponding touch events).
+ * While dragging the current column can be updated, and on mouse up the width is confirmed.
+ * By passing callbacks the behavior for updating and confirming width can be modified.
+ */
+const ColumnAdjuster = ({
+  columnWidth,
+  keyValue,
+  isLastColumn,
+  updateWidthCallback,
+  confirmWidthCallback,
+  handleBlur,
+}: ColumnAdjusterProps) => {
+  const tempWidth = useMemo(() => ({ initWidth: columnWidth, columnWidth, initX: 0 }), [columnWidth]);
+
+  // Note that deltaWidth is the change since you started the resize
+  const updateWidth = (deltaWidth: number) => {
+    // TODO: table and pivot table min widths are different at this point
+    const adjustedWidth = Math.max(tempWidth.initWidth + deltaWidth, ColumnWidthValues.PixelsMin);
+    tempWidth.columnWidth = adjustedWidth;
+    updateWidthCallback(adjustedWidth);
+  };
+
+  const confirmWidth = () => {
+    if (tempWidth.columnWidth !== tempWidth.initWidth) {
+      const newWidthData = { type: ColumnWidthType.Pixels, pixels: tempWidth.columnWidth };
+      confirmWidthCallback(newWidthData);
+    }
+  };
+
+  // ----- Mouse -----
+  const mouseMoveHandler = (evt: MouseEvent) => {
+    const deltaWidth = evt.clientX - tempWidth.initX;
+    updateWidth(deltaWidth);
+  };
+
+  const mouseUpHandler = (evt: MouseEvent) => {
+    preventDefaultBehavior(evt);
+    document.removeEventListener('mousemove', mouseMoveHandler);
+    document.removeEventListener('mouseup', mouseUpHandler);
+
+    confirmWidth();
+  };
+
+  const mouseDownHandler = (evt: React.MouseEvent) => {
+    evt.stopPropagation();
+    document.addEventListener('mousemove', mouseMoveHandler);
+    document.addEventListener('mouseup', mouseUpHandler);
+
+    tempWidth.initX = evt.clientX;
+  };
+
+  const doubleClickHandler = () => confirmWidthCallback({ type: ColumnWidthType.FitToContent });
+
+  // ----- Touch -----
+  const touchMoveHandler = (evt: TouchEvent) => {
+    const deltaWidth = evt.touches[0].clientX - tempWidth.initX;
+    updateWidth(deltaWidth);
+  };
+
+  const touchEndHandler = (evt: TouchEvent) => {
+    evt.preventDefault();
+    evt.stopPropagation();
+    document.removeEventListener('touchmove', touchMoveHandler);
+    document.removeEventListener('touchend', touchEndHandler);
+
+    confirmWidth();
+  };
+
+  const touchStartHandler = (evt: React.TouchEvent) => {
+    if (evt.touches.length !== 1) return;
+
+    evt.stopPropagation();
+    document.addEventListener('touchmove', touchMoveHandler);
+    document.addEventListener('touchend', touchEndHandler);
+
+    tempWidth.initX = evt.touches[0].clientX;
+  };
+
+  // ----- Keyboard -----
+  const KeyDownHandler = handleBlur
+    ? (event: React.KeyboardEvent) => {
+        if (event.key === KeyCodes.LEFT || event.key === KeyCodes.RIGHT) {
+          preventDefaultBehavior(event);
```

Comment:
this line being used in all cases. I think it's better to keep it out of cases unless you want to default behaviour bubbles up for other key strokes

Reply:
for all the keys that actually does anything for the column width, yes. But we don't want to block all other keys